commands/operator-sdk/cmd/scorecard/resource_handler.go

```diff
@@ -0,0 +1,294 @@
+				restMapper.Reset()
+				_, err := restMapper.RESTMappings(obj.GetObjectKind().GroupVersionKind().GroupKind())
+				if err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			err = runtimeClient.Create(context.TODO(), obj)
+			if err != nil {
+				return err
+			}
+		}
+		addResourceCleanup(obj, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+	}
+	return nil
+}
+
+// createKubeconfigSecret creates the secret that will be mounted in the operator's container and contains
+// the kubeconfig for communicating with the proxy
+func createKubeconfigSecret() error {
+	kubeconfigMap := make(map[string][]byte)
+	kc, err := proxyConf.Create(metav1.OwnerReference{Name: "scorecard"}, "http://localhost:8889", SCConf.Namespace)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.Remove(kc.Name()); err != nil {
+			log.Errorf("Failed to delete generated kubeconfig file: (%v)", err)
+		}
+	}()
+	kc, err = os.Open(kc.Name())
+	if err != nil {
+		return err
+	}
+	kcBytes, err := ioutil.ReadAll(kc)
+	if err != nil {
+		return err
+	}
+	kubeconfigMap["kubeconfig"] = kcBytes
+	kubeconfigSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "scorecard-kubeconfig",
+			Namespace: SCConf.Namespace,
+		},
+		Data: kubeconfigMap,
+	}
+	err = runtimeClient.Create(context.TODO(), kubeconfigSecret)
+	if err != nil {
+		return err
+	}
+	addResourceCleanup(kubeconfigSecret, types.NamespacedName{Namespace: kubeconfigSecret.GetNamespace(), Name: kubeconfigSecret.GetName()})
+	return nil
+}
+
+// addMountKubeconfigSecret creates the volume mount for the kubeconfig secret
+func addMountKubeconfigSecret(dep *appsv1.Deployment) {
+	// create mount for secret
+	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, v1.Volume{
+		Name: "scorecard-kubeconfig",
+		VolumeSource: v1.VolumeSource{Secret: &v1.SecretVolumeSource{
+			SecretName: "scorecard-kubeconfig",
+			Items: []v1.KeyToPath{{
+				Key:  "kubeconfig",
+				Path: "config",
+			}},
+		},
+		},
+	})
+	for index := range dep.Spec.Template.Spec.Containers {
+		// mount the volume
+		dep.Spec.Template.Spec.Containers[index].VolumeMounts = append(dep.Spec.Template.Spec.Containers[index].VolumeMounts, v1.VolumeMount{
+			Name:      "scorecard-kubeconfig",
+			MountPath: "/scorecard-secret",
+		})
+		// specify the path via KUBECONFIG env var
+		dep.Spec.Template.Spec.Containers[index].Env = append(dep.Spec.Template.Spec.Containers[index].Env, v1.EnvVar{
+			Name:  "KUBECONFIG",
+			Value: "/scorecard-secret/config",
+		})
+	}
+}
+
+// addProxyContainer adds the container spec for the scorecard-proxy to the deployment's podspec
+func addProxyContainer(dep *appsv1.Deployment) {
+	var pullPolicy v1.PullPolicy
+	if SCConf.ProxyPullPolicy == "Always" {
+		pullPolicy = v1.PullAlways
+	} else if SCConf.ProxyPullPolicy == "Never" {
+		pullPolicy = v1.PullNever
+	} else if SCConf.ProxyPullPolicy == "PullIfNotPresent" {
```

Comment:
Just a nit but you can use the PullPolicy constants defined upstream for this:
https://github.com/kubernetes/api/blob/b503174bad5991eb66f18247f52e41c3258f6348/core/v1/types.go#L1961-L1968

Reply:
Sorry ignore the above comment as you're already using the [PullPolicy constants](https://github.com/kubernetes/api/blob/b503174bad5991eb66f18247f52e41c3258f6348/core/v1/types.go#L1962-L1967).

Is there a reason why we're comparing and checking the `proxy-pull-policy` flag with the string values, rather than defining `SCConf.ProxyPullPolicy` as type `v1.PullPolicy` and comparing that with the constants directly?

Reply:
We declare the `proxy-pull-policy` flag as a string to cobra. If define `SCConf.ProxyPullPolicy` as a `v1.PullPolicy` that results in a syntax error. There is also a syntax error if I use the current `SCConf.ProxyPullPolicy` in the spec directly, which is why I need to use the `if` statements.

Reply:
This might look cleaner with a switch I think.

Reply:
@shawn-hurley Yeah, I'll make that change